src/Adapters/Auth/index.js

```diff
@@ -56,15 +59,40 @@ function authDataValidator(adapter, appIds, options) {
   }
 }
 
+function validateAdapter(adapter) {
```

Comment:
I'm not sure why all of this was extracted, I seems to be a copy paste from the adapter loader.

The original implementation, that just validates the presence of both methods as enough IMHO.

Either:

- Revert to the simple validation
- Extract the validator provided [here](https://github.com/parse-community/parse-server/blob/9de4b8b2a7cf4487271f3eeef84b307d53c772f4/src/Controllers/AdaptableController.js#L56) into it's own module and refactor both places.

At this point, it's overkill so please revert and add a TODO for later.

It's also untested :)